Terminal.Gui/Views/TreeViewTextFilter.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// <see cref="ITreeViewFilter{T}"/> implementation which searches the
+	/// <see cref="TreeView{T}.AspectGetter"/> of the model for the given
+	/// <see cref="Text"/>.
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	public class TreeViewTextFilter<T> : ITreeViewFilter<T> where T : class {
+		readonly TreeView<T> forTree;
+
+		/// <summary>
+		/// Creates a new instance of the filter for use with <paramref name="forTree"/>.
+		/// Set <see cref="Text"/> to begin filtering.
+		/// </summary>
+		/// <param name="forTree"></param>
+		/// <exception cref="ArgumentNullException"></exception>
+		public TreeViewTextFilter (TreeView<T> forTree)
+		{
+			this.forTree = forTree ?? throw new ArgumentNullException (nameof (forTree));
```

Comment:
And get rid of `this.`

Reply:
Done in [42e32b1](https://github.com/gui-cs/Terminal.Gui/pull/2599/commits/42e32b194b750c1fc23ec8bcb4ba819676bf740d)